backend_tsc/src/entitites/site.ts

```diff
@@ -0,0 +1,36 @@
+import {Schema,Types,Model,model} from "mongoose"
+import {IMongo_Entity} from "./mongo_entity";
+import Catalogue from "./catalogue";
+import Region from "./region";
+import ProjectilePoint, {IProjectilePoint} from "./projectile_point";
+import {cascade_related} from "../utilities/trigger_factory";
+
+
+//Schema
+export interface ISite extends IMongo_Entity{
+    name: string,
+    description: string,
+    catalogue_id: Types.ObjectId
+    region_id: Types.ObjectId
+}
+
+type SiteModal = Model<ISite>;
+const siteSchema = new Schema<ISite,SiteModal>({
+    name:{type:String, required:true},
+    description:{type:String, required:false},
+    catalogue_id:{type: Schema.Types.ObjectId, ref:"Catalogue",required:true},//TODO using actual models should not break...
```

Comment:
Not really sure what this means. Is using actual models non-normal? Why "shouldn't it break"?

Reply:
Right! So rather than using `"Catalogue"` for example as the ref parameter, we could use the actual model that is:
```ts
import Catalogue from "./Catalogue" //(entities/catalogue)

...
{
catalogue_id:{type:X, ref:Catalogue, ...}
}
```
It has no effect on functionality which we use. This one works consistently, so I switched them all to use the collection-name string. Could probably remove the TODO.